src/server/lib/LineReader.ts

```diff
@@ -3,24 +3,17 @@
  * Licensed under the MIT License. See License.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
 
-import { URL } from 'url';
+import {
+    TextDocument,
+} from 'vscode-languageserver-textdocument';
 
-// eslint-disable-next-line @typescript-eslint/no-var-requires
-const lineByLine = require('n-readlines');
-
-export function getEditedLineContent(rowIndex: number, fileUrl: URL): string {
-    const liner = new lineByLine(fileUrl);
-    let line = liner.next();
-    let lineNumber = 0;
-    let userEditedLine = '';
-
-    while (line) {
-        if (lineNumber == rowIndex) {
-            userEditedLine = line.toString('ascii');
-            break;
+export function getEditedLineContent(rowIndex: number, textDocument: TextDocument): string {
+    const lines = textDocument.getText().split(/\r?\n/g);
```

Comment:
since textDoc is now assigned event driven, it's possible for it to be undefined -> use optional chaining .?  to protect here and the for loop in line 13

Reply:
consider changing the func sig to: textDocument: TextDocument | undefined to let the TS compiler type assert properly